compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/SemanticAnalyzer.java

```diff
@@ -513,6 +514,17 @@ public void visit(BLangTypeDefinition typeDefinition) {
 
         if (typeDefinition.flagSet.contains(Flag.ENUM)) {
             ((BEnumSymbol) typeDefinition.symbol).addAnnotations(annotSymbols);
+            HashSet<String> enumElements = new HashSet<String>();
+            BLangUnionTypeNode bLangUnionTypeNode = (BLangUnionTypeNode)  typeDefinition.typeNode;
+            for (int j = bLangUnionTypeNode.memberTypeNodes.size() - 1; j >= 0; j--) {
```

Comment:
Any particular reason why we iterate in reverse order, if we can iterate on regular order we can simply replace this with a simple for loop

Reply:
Enum union type list is in reverse order of the symbols present in particular enum. Therefore, we have to travel the list in reverse order to get the actually re-declared symbols.